spring-cloud-commons/src/main/java/org/springframework/cloud/client/DefaultServiceInstance.java

```diff
@@ -57,6 +59,23 @@ public URI getUri() {
 		return this.metadata;
 	}
 
+	@Override
+	public void putMetadata(String key, String value) {
+		this.metadata.put(key, value);
+	}
+
+	@Override
+	public void setMetadata(Map<String, String> metadata) {
```

Comment:
I took the approach of removing all the old fields and adding them because `metadata` is `final` for this class.

Removing `final` from the field seemed more problematic because of the `@RequiredArgsConstructor`.